File: src/main/java/application/history/history/presentation/interfaces/api/DeleteHistoryController.java
package application.history.history.presentation.interfaces.api;

import application.history.history.application.services.DeleteHistoryService;
import org.springframework.format.annotation.DateTimeFormat;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

import java.time.LocalDate;

@RestController
public class DeleteHistoryController {
    private final DeleteHistoryService service;

    public DeleteHistoryController(DeleteHistoryService deleteHistoryService) {
        this.service = deleteHistoryService;
    }

    @DeleteMapping("/histories/{date}")
    public ResponseEntity<Object> delete(@PathVariable("date") @DateTimeFormat(pattern = "yyyy-MM-dd") LocalDate date) {
        this.service.run(date);

        return ResponseEntity.noContent().build();
    }
}

File: src/main/kotlin/com/example/history/presentation/interfaces/api/DeleteHistoryController.kt
package com.example.history.presentation.interfaces.api

import com.example.history.application.services.FindHistoryUseCase
import com.example.history.application.services.RemoveHistoryUseCase
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/histories")
class DeleteHistoryController(private val findHistoryService: FindHistoryUseCase, private val removeHistoryService: RemoveHistoryUseCase) {

    @DeleteMapping("/{id}")
    fun delete(@PathVariable("id") id: String): ResponseEntity<Any> {
        removeHistoryService.remove(findHistoryService.findById(id))

        return ResponseEntity.noContent().build()
    }
}
